### PR TITLE
Migrate `bootstrap-hot-loader` to TypeScript

### PR DESCRIPTION
**Pull request by ronami**
### Summary

Pretty straight-forward. See #1401 for details.

### CODE REVIEW

File: packages/bootstrap-hot-loader/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "src",
+    "outDir": "build"
+  },
+  "include": [
+    "src/**/*",
+    "src/**/*.json"
```
Comment (ranyitz) [inline packages/bootstrap-hot-loader/tsconfig.json]: I've asked myself: why is this needed?

Answer to myself:

> If a segment of a glob pattern includes only `*` or `.*`, then only files with supported extensions are included (e.g. `.ts`, `.tsx`, and `.d.ts` by default with `.js` and `.jsx` if `allowJs` is set to true).

https://www.typescriptlang.org/docs/handbook/tsconfig-json.html

Comment (ronami) [reply packages/bootstrap-hot-loader/tsconfig.json]: Also: https://github.com/Microsoft/TypeScript/issues/25636

Comment (ranyitz) [reply packages/bootstrap-hot-loader/tsconfig.json]: @ronami That was the next question 😆
